Userland/Libraries/LibGUI/Dialog.cpp

```diff
@@ -28,18 +28,25 @@ int Dialog::exec()
     VERIFY(!m_event_loop);
     m_event_loop = make<Core::EventLoop>();
 
-    auto desktop_rect = Desktop::the().rect();
+    bool has_parent_window = parent() && is<Window>(parent());
+
+    size_t screen_index = Desktop::the().main_screen_index();
+    if (has_parent_window) {
+        screen_index = static_cast<Window*>(parent())->containing_screen_index();
```

Comment:
Overwriting `screen_index` seems wrong here. The value returned by `Desktop::the().main_screen_index()` is the screen where the taskbar is located on, which is information we should take into account when centering a window (there's no taskbar on other screens).